### PR TITLE
Feature/typescript

### PR DESCRIPTION
**Pull request by bigsergey**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,14 +1,32 @@
...
-  "main": "index.js",
+  "main": "lib/index.js",
+  "files": [
+    "lib/**/*"
+  ],
   "scripts": {
-    "start": "node index.js"
+    "start": "npm run build:live",
+    "build": "tsc -p .",
+    "build:live": "nodemon --watch 'src/**/*.ts' --exec 'ts-node' src/index.ts",
```
Comment (erykpiast) [inline package.json]: why not `tsc --watch --incremental`?

**Comment by bigsergey** [comment]
Fixes #3

File: src/lint-files.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+function lintFile(file: string): string | undefined {
+  const extensionName = path.extname(file);
+  const basename = path.basename(file, extensionName);
+  const isValid = kebabCase(basename) === basename;
+
+  return isValid ? undefined : `${file} has wrong name.`;
+}
+
+export default function lintFiles(files: string[]): string[] {
+  return files.reduce((acc: string[], file: string): string[] => {
```
Comment (erykpiast) [inline src/lint-files.ts]: Is it required by linter to explicitly type everything? Both `acc` and `file` should derive types I suppose.

The same goes for `lintResult`.

Comment (erykpiast) [inline src/lint-files.ts]: `reduce` -> `flatMap`. The result should be the same, but the intention more clear.

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,70 @@
+{
+  "compilerOptions": {
+    /* Basic Options */
+    // "incremental": true,                   /* Enable incremental compilation */
+    "target": "es5",                          /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019' or 'ESNEXT'. */
```
Comment (erykpiast) [inline tsconfig.json]: you require node >=12.10, ES2018 is enough :)

File: tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,70 @@
+{
```
Comment (erykpiast) [inline tsconfig.json]: all these comments look a little odd.

File: tsconfig.json (hunk 3 above)
Comment (bigsergey) [reply tsconfig.json]: 👍 you are right!

File: package.json (hunk 1 above)
Comment (bigsergey) [reply package.json]: nodemon has `--exec` option which is useful now. I am going to remove it later.

File: src/lint-files.ts (hunk 2 above)
Comment (bigsergey) [reply src/lint-files.ts]: done

Comment (bigsergey) [reply src/lint-files.ts]: I thought  it had to be this way

File: src/lint-files.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  const extensionName = path.extname(file);
+  const basename = path.basename(file, extensionName);
+  const isValid = kebabCase(basename) === basename;
+
+  return isValid ? undefined : `${file} has wrong name.`;
+}
+
+export default function lintFiles(files: string[]): string[] {
+  return files.flatMap(file => {
+    const lintResult = lintFile(file);
```
Comment (adrianmroz) [inline src/lint-files.ts]: What about `""`?

Comment (bigsergey) [reply src/lint-files.ts]: done!

File: tsconfig.json (hunk 4 above)
Comment (bigsergey) [reply tsconfig.json]: done!

File: .eslintrc.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+module.exports =  {
+  parser:  '@typescript-eslint/parser',  // Specifies the ESLint parser
+  extends:  [
+    'plugin:@typescript-eslint/recommended',  // Uses the recommended rules from the @typescript-eslint/eslint-plugin
```
[the remark this answers is not in the document]
Comment (bigsergey) [reply .eslintrc.js]: nice catch, fixed

**Pull request merged**